src/librustc_codegen_ssa/meth.rs

```diff
@@ -87,15 +87,15 @@ pub fn get_vtable<'tcx, Cx: CodegenMethods<'tcx>>(
 
     let methods = methods.cloned().map(|opt_mth| {
         opt_mth.map_or(nullptr, |(def_id, substs)| {
-            cx.get_fn_addr(
-                ty::Instance::resolve_for_vtable(
-                    cx.tcx(),
-                    ty::ParamEnv::reveal_all(),
-                    def_id,
-                    substs,
-                )
-                .unwrap(),
+            let mut instance = ty::Instance::resolve_for_vtable(
+                cx.tcx(),
+                ty::ParamEnv::reveal_all(),
+                def_id,
+                substs,
             )
+            .unwrap();
+            instance.polymorphize(cx.tcx());
```

Comment:
By-value `Instance::polymorphize` would make code like this nicer, since it'd just be an added `.polymorphize()`.

Reply:
I've fixed this in ac01dac.